lib/src/block_parser.dart

```diff
@@ -54,10 +54,34 @@ class BlockParser {
   /// The markdown document this parser is parsing.
   final Document document;
 
+  /// The enabled block syntaxes. To turn a series of lines into blocks, each of
+  /// these will be tried in turn. Order matters here.
+  final List<BlockSyntax> blockSyntaxes = [];
+
   /// Index of the current line.
   int _pos;
 
-  BlockParser(this.lines, this.document) : _pos = 0;
+  /// The collection of built-in block parsers.
+  final List<BlockSyntax> standardBlockSyntaxes = const [
+    const EmptyBlockSyntax(),
+    const BlockHtmlSyntax(),
+    const SetextHeaderSyntax(),
+    const HeaderSyntax(),
+    const CodeBlockSyntax(),
+    const BlockquoteSyntax(),
+    const HorizontalRuleSyntax(),
+    const UnorderedListSyntax(),
+    const OrderedListSyntax(),
+    const ParagraphSyntax()
+  ];
+
+  BlockParser(this.lines, this.document) {
+    _pos = 0;
+    if (document.extensions.contains('fenced-code-blocks')) {
```

Comment:
Stringly typed APIs make me sad. How about an enum or make the extensions actual objects (i.e. similar to how the extension for a block syntax *is* the object that supports that syntax)?